portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -28,37 +34,73 @@
 @WebServlet("/comments")
 public final class CommentServlet extends HttpServlet {
 
-    private List<Comment> comments;
-
-    @Override
-    public void init() {
-        comments = new ArrayList<>();
-    }
-
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Query commentQuery = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery commentResults = datastore.prepare(commentQuery);
+
+        List<Comment> comments = new ArrayList<>();
+
+        for (Entity commentEntity : commentResults.asIterable()) {
+            long id = commentEntity.getKey().getId();
+            String commenter = (String) commentEntity.getProperty("commenter");
+            String message = (String) commentEntity.getProperty("message");
+
+            Comment comment = new Comment(id, commenter, message);
+
+            Query replyQuery = new Query("Reply").addSort("timestamp", SortDirection.DESCENDING);
+            PreparedQuery replyResults = datastore.prepare(replyQuery);
+
+            for (Entity replyEntity : replyResults.asIterable()) {
+                long parentId = (long) replyEntity.getProperty("parentId");
+                if(parentId == id){
```

Comment:
I'm having trouble filtering queries. So, here I've added an if statement instead. Previously, line 54 was
`Query replyQuery = new Query("Reply").setFilter(new FilterPredicate("parentId", FilterOperator.EQUAL, id)).addSort("timestamp", SortDirection.DESCENDING);`

This worked in my dev server but returned a 500 error in my live server. 
I also tried referencing [this page about filters](https://cloud.google.com/datastore/docs/concepts/queries#filters). But this did not work for me in my live nor dev server. 

The current approach works fine. It's just that I was wondering if I could filter the query instead as this requires me to iterate through each reply multiple times.